packages/dds/tree/src/simple-tree/types.ts

```diff
@@ -212,6 +211,24 @@ export abstract class TreeNode implements WithType {
 	}
 }
 
+/**
+ * Returns a schema for a value if the value is a {@link TreeNode}.
+ *
+ * Returns undefined for other values.
+ * @remarks
+ * Does not give schema for a {@link TreeLeafValue}.
+ */
+export function tryGetTreeNodeSchema<T>(
```

Comment:
this logic was factored out of treeNodeApi.ts's more general tryGetSchema so that the rest of that file could move to the "api" layer.